Use string to request default solver parameters

Patol75 opened the pull request:
Default iterative and direct Stokes solver parameters can now be specified using eponymous strings.

drhodrid commented:
Going to tag @stephankramer and @angus-g for this one - just in case I miss something subtle. However, it looks simple and good to me.

drhodrid commented:
Could you add a short test of the new functionality?

angus-g commented:
> Could you add a short test of the new functionality?

Probably a good candidate for just a unit test here.

I think the proposed idea is good and sensibly-implemented. I guess the `match/case` construct is Python 3.10+, which we kind of already require because of the type annotations anyway. I think it's fair to say that Python 3.10 is easily available, but it is indeed not always the _default_, which can cause issues.

drhodrid commented:
Thomas please add a test, and then this is good to go.

Patol75 commented:
Thank you for requesting a test; my implementation did not work! Not sure how that happened. Anyway, I fixed it and part of the logic in `stokes_integrator.py`; let me know if everything looks reasonable to you. I added a test, which passes locally. Hopefully, the CI will confirm. One small issue I spotted is that `fd.interpolate` is still used in some of the tests despite its deprecation status.

angus-g commented on tests/test_stokes_solver_configuration.py:
```diff
@@ -0,0 +1,93 @@
+            solver_parameters=solver_parameters,
+        )
+
+        np.testing.assert_equal(stokes_solver.solver_parameters, expected_value)
+
+    np.testing.assert_raises(
+        ValueError,
+        StokesSolver,
+        stokes_function,
+        temperature,
+        approximation,
+        solver_parameters="",
+    )
```
I guess `assert_equal` is fine enough here (although it doesn't need any of the np-specific handling), but I think it would be clearer to use a regular `with pytest.raises`. And in that case, might as well not use np in this file at all?

Patol75 replied:
Thank you, Angus. I did not realise that `pytest` is very capable to handle verbosely plain `assert`. Should be good to go now.

angus-g reviewed (commented):
Test looks good, this is just about ready to go

angus-g reviewed (approved):
Looks good, thanks!
